shared/bls/bls.go

```diff
@@ -64,145 +70,195 @@ func PublicKeyFromBytes(pub []byte) (*PublicKey, error) {
 		return cv.Value().(*PublicKey).Copy(), nil
 	}
 	b := bytesutil.ToBytes48(pub)
-	k, err := g1.DeserializePublicKey(b)
+	g1Elems := bls12.NewG1(nil)
+	p, err := g1Elems.FromCompressed(b[:])
 	if err != nil {
 		return nil, errors.Wrap(err, "could not unmarshal bytes into public key")
 	}
-	pk := &PublicKey{val: k}
-	pubkeyCache.Set(string(pub), pk.Copy(), 48*time.Hour)
-	return pk, nil
+	copiedVal := &bls12.PointG1{}
+	g1Elems.Copy(copiedVal, p)
+	pubkeyCache.Set(string(pub), copiedVal, 48*time.Hour)
+	return &PublicKey{p: p}, nil
 }
 
 // SignatureFromBytes creates a BLS signature from a LittleEndian byte slice.
 func SignatureFromBytes(sig []byte) (*Signature, error) {
 	if featureconfig.FeatureConfig().SkipBLSVerify {
 		return &Signature{}, nil
 	}
-	b := bytesutil.ToBytes96(sig)
-	s, err := g1.DeserializeSignature(b)
+	s, err := bls12.NewG2(nil).FromCompressed(sig)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not unmarshal bytes into signature")
 	}
-	return &Signature{val: s}, nil
+	return &Signature{s: s}, nil
 }
 
 // PublicKey obtains the public key corresponding to the BLS secret key.
 func (s *SecretKey) PublicKey() *PublicKey {
-	return &PublicKey{val: g1.PrivToPub(s.val)}
+	p := &bls12.PointG1{}
+	bls12.NewG1(nil).MulScalar(p, &bls12.G1One, s.p)
+	return &PublicKey{p: p}
 }
 
-// Sign a message using a secret key - in a beacon/validator client,
+// Sign a message using a secret key - in a beacon/validator client.
 func (s *SecretKey) Sign(msg []byte, domain uint64) *Signature {
 	if featureconfig.FeatureConfig().SkipBLSVerify {
 		return &Signature{}
 	}
-	b := make([]byte, 8)
-	binary.LittleEndian.PutUint64(b, domain)
-	sig := g1.SignWithDomain(bytesutil.ToBytes32(msg), s.val, bytesutil.ToBytes8(b))
-	return &Signature{val: sig}
+	g2 := bls12.NewG2(nil)
+	b := [8]byte{}
+	binary.LittleEndian.PutUint64(b[:], domain)
+	signature := g2.MapToPoint(HashWithDomain(bytesutil.ToBytes32(msg), b))
+	g2.MulScalar(signature, signature, s.p)
+	return &Signature{s: signature}
 }
 
 // Marshal a secret key into a LittleEndian byte slice.
 func (s *SecretKey) Marshal() []byte {
-	k := s.val.Serialize()
-	return k[:]
+	keyBytes := s.p.Bytes()
+	if len(keyBytes) < 32 {
+		emptyBytes := make([]byte, 32-len(keyBytes))
+		keyBytes = append(emptyBytes, keyBytes...)
+	}
+	return keyBytes
 }
 
 // Marshal a public key into a LittleEndian byte slice.
 func (p *PublicKey) Marshal() []byte {
-	k := p.val.Serialize()
-	return k[:]
+	return bls12.NewG1(nil).ToCompressed(p.p)
 }
 
-// Copy the public key to a new pointer reference.
+// Copy copies the public key and returns it.
```

Comment:
Can you revert this comment? 

```suggestion
// Copy the public key to a new pointer reference.
```